tests/compile-fail/never_say_never.rs

```diff
@@ -4,7 +4,7 @@
 fn main() {
     let y = &5;
     let x: ! = unsafe {
-        *(y as *const _ as *const !) //~ ERROR entered unreachable code
+        *(y as *const _ as *const !) //~ ERROR tried to access a dead local variable
```

Comment:
But I don't see *how* it's possible to catch `!` here, almost seems like an unrelated bug :/.